main.tf

```diff
@@ -98,6 +87,38 @@ resource "aws_s3_bucket" "quortex" {
   }
 }
 
+resource "aws_s3_bucket_acl" "quortex" {
+  for_each = var.buckets
+
+  bucket = aws_s3_bucket.quortex[each.value].id
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "quortex" {
+  for_each = var.expiration != null ? var.buckets : []
+
+  bucket = aws_s3_bucket.quortex[each.value].id
+  rule {
+    id     = "expiration"
+    status = "Enabled"
+    expiration {
+      days = var.expiration.expiration_days
+    }
+  }
+}
+
+# Set minimal encryption on buckets
+resource "aws_s3_bucket_server_side_encryption_configuration" "quortex" {
+  for_each = var.enable_bucket_encryption ? var.buckets : toset([])
```

Comment:
Is the `toset` function required ? It is not used in the `aws_s3_bucket_lifecycle_configuration.quortex` part.

Reply:
yes, indeed it was missing on lifecycle conf